midend/flattenHeaders.cpp

```diff
@@ -128,8 +128,10 @@ const IR::Node* ReplaceHeaders::postorder(IR::Member* expression) {
                   "FH cannot find replacement for %s in type %s", prefix, h);
     }
 
-    result = new IR::Member(e, newFieldName);
+    result = new IR::Member(expression->type, e, newFieldName);
```

Comment:
This may not be sufficient. There may be other expressions that involve the struct whose type may change.
Also, in general expression->type is stale; it is not maintained by the compiler. If you want to set the type use the type in the typeMap.

Reply:
I see that you ask the typechecker to set these types; maybe the type won't be stale at this point.

Reply:
For example, the type of any method that handles this kind of structures will change too.